projects/plugins/jetpack/modules/masterbar/admin-menu/load.php

```diff
@@ -7,6 +7,33 @@
 
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
+/**
+ * Checks whether the navigation customizations should be performed for the given class.
+ *
+ * @param string $admin_menu_class Class name.
+ * @return bool
+ */
+function should_customize_nav( $admin_menu_class ) {
+	// Make sure the class extends the base admin menu class.
+	if ( ! is_subclass_of( $admin_menu_class, Base_Admin_Menu::class ) ) {
+		return false;
+	}
+
+	$is_api_request = defined( 'REST_REQUEST' ) && REST_REQUEST || 0 === strpos( $_SERVER['REQUEST_URI'], '/?rest_route=%2Fwpcom%2Fv2%2Fadmin-menu' );
+
+	// No nav customizations on WP Admin of Atomic sites when SSO is disabled.
+	if ( Atomic_Admin_Menu::class === $admin_menu_class && ! $is_api_request && ! \Jetpack::is_module_active( 'sso' ) ) {
```

Comment:
A bit of a nitpick, I think we could do a small micro-optimization here to avoid executing strpos on each request for WPCOM.

We could move `$is_api_request` variable at the end and have a condition like 
```
if ( Atomic_Admin_Menu::class !== $admin_menu_class && Jetpack_Admin_Menu::class !== $admin_menu_class ) {
    return true;
}

....
```

Reply:
`strpos` should take only microseconds, so the performance shouldn't be affected. For this case, I'd prefer to avoid the micro-optimization in order to keep the code a bit more readable.